src/components/ViewItem.vue

```diff
@@ -1,15 +1,138 @@
+    <div id="item-display" v-if="STATUS != 'loading' && !!item">
+      <div class="row">
+        <label class="col-md-4" for="title">Title:</label>
+        <input type="text" ref="title" id="title" :value="item.title"
+          :disabled="!isEditing" :class="{view: !isEditing}">
+      </div>
+      <div class="row">
+        <label class="col-md-4" for="content">Content:</label>
+        <textarea  class="col-md-5" ref="content" id="content" :value="item.content"
+          :disabled="!isEditing" :class="{view: !isEditing}"></textarea>
+      </div>
+      <div class="row">
+        <label class="col-md-4" for="priority">Priority:</label>
+        <select class="col-md-1" ref="priority" id="priority" :value="item.priority"
+          :disabled="!isEditing" :class="{view: !isEditing}">
+          <option value="0">Low</option>
+          <option value="1">Middle</option>
+          <option value="2">High</option>
+        </select>
+      </div>
+      <div class="row">
+        <label class="col-md-4" for="is_done">Done:</label>
+        <input type="checkbox" ref="is_done" id="is_done" :checked="item.is_done"
+          :disabled="!isEditing" :class="{view: !isEditing}">
+      </div>
+      <div class="row">
+        <div class="col-md-3 offset-md-3 form-group">
+          <span class="col-md-1">
+            <button @click="isEditing = true" v-if="!isEditing" class="btn btn-primary">
+              Edit
+            </button>
+          </span>
+          <span class="col-md-1">
+            <button @click="saveItem()" v-if="isEditing" class="btn btn-success">
+              Save
+            </button>
+          </span>
+
+          <button v-if="isEditing" @click="isEditing = false" class="btn btn-default">
+            Cancel
+          </button>
+        </div>
+        <div class="col-md-2 form-group">
+          <button @click="deleteItem()" v-if="!isEditing" class="btn btn-danger">
+            Delete
+          </button>
+        </div>
+      </div>
+    </div>
+    <div class="spinner" v-if="STATUS === 'loading'"></div>
   </div>
 </template>
 
 <script>
+import { mapGetters } from 'vuex'
+import { GET_ITEM, UPDATE_ITEM, DELETE_ITEM, STATUS } from '../constants'
+
 export default {
   name: 'ViewItem',
+  props: [
+    'id'
+  ],
+  computed: {
+    item () {
+      return this.$store.getters[GET_ITEM](this.$route.params.id)
+    },
+    ...mapGetters([STATUS])
+  },
   data () {
     return {
-      item: '\'item\' prop of \'data\' object'
+      isEditing: false
     }
+  },
+  methods: {
+    edit () {
+      this.isEditing = true
+    },
+    itemChanged (updatedItem) {
+      return updatedItem.title !== this.item.title ||
+        updatedItem.content !== this.item.content ||
+        updatedItem.priority !== this.item.priority ||
+        updatedItem.is_done !== this.item.is_done
+    },
+    saveItem () {
+      const updatedItem = {
+        id: this.item.id,
+        title: this.$refs['title'].value,
```

Comment:
Ovde umesto refs, nisam sigurna zasto nisi koristio `v-model` na tim poljima u formi?

Reply:
`v-model` bih morao da vezem za neki objekat, a meni je `item` u ovoj komponenti `computed property`, kojeg "vadim" iz `state.items` niza.
Mogao sam ili da uvedem neki objekat koji ce biti "zamjena" za item, pa da njega bindujem, ili da koristim `refs`. Druga opcija mi se ucinila jednostavnija (odnosno sa manje koda), pa sam to odabrao :)